### PR TITLE
feat: exposes gotoPeriod method to top level, removes unused event parameter

### PR DESCRIPTION
**Pull request by MarcelR1998**
## Checklist

Please put "X" in the below checkboxes that apply::

- [ ] If committing a bugfix, I have tested it in different browsers (Chrome, Firefox, Safari). 
- [x] If committing a new feature, I have first submitted an issue (Please note: you are free to open PRs for non-issued features, but opening an issue increases your chance of a successful PR). 
- [ ] If committing a new feature, I have also written an appropriate test suite for it. 

I have tested the following:  

- [ ] Qalendar component in month mode. 
- [ ] Qalendar component in week mode. 
- [ ] Qalendar component in day mode. 
- [ ] All of the above modes on emulated mobile view. 
- [ ] Dragging and dropping events. 
- [ ] Resizing events in day/week modes. 
- [ ] Clicking events to open event dialog. 

## This PR solves the following problem**. 
Allows for more customization of the calendar. A fully custom header can be achieved by hiding the default one, calling for example `@click="$refs.calendar.mode = 'month'` to change mode, and with my pull request, utilize the existing goToPeriod method to change period rather than having to create a method that modifies selected-date.

I have run the existing Vitest suite, and I guess an improvement could be to create a test along with buttons in the same file as the Qalendar element, with the test running the method through the ref by clicking

## How to test this PR**. 

For example:  
1. Add a template ref to wherever Qalendar is used
2. Create new buttons, with a click event that fires `$refs.<refname>.goToPeriod('next')` and `$refs.<refname>.goToPeriod('previous')`

### CODE REVIEW

**Comment by tomosterlund** [comment]
Hi @MarcelR1998. Thanks for opening this PR 👍  I'll have a look at it after work tonight

File: src/components/header/Header.vue [hunk 1]
```diff
@@ -188,7 +188,7 @@ export default defineComponent({
       this.$emit('updated-period', value);
     },
 
-    goToPeriod(event: MouseEvent, direction: 'previous' | 'next') {
```
Comment (tomosterlund) [inline src/components/header/Header.vue:191]: 👍 

very good. Thanks for cleaning up

**Review by tomosterlund — APPROVED** [review]
Just added a unit test, but otherwise everything looks good to me!

**Comment by MarcelR1998** [comment]
:partying_face: :partying_face: <3<3

**Pull request merged**